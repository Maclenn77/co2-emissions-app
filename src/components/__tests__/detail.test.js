/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import Detail from '../detail';

const mockStore = configureMockStore();

const store = mockStore({
  co2Emissions: {
    0: {
      id: 0,
      iso: 'mex',
      label: 'Mexico',
      date: 2018,
      co2Emissions: 6.0000,
    },
  },
});

it('renders correctly', () => {
  const mockDetail = {
    iso: 'MCK',
    label: 'Mock Country',
    date: 2018,
    co2Emissions: 6.0000,
  };
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <Detail year={mockDetail} />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
