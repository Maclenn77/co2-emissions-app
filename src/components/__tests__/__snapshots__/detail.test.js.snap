// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="d-flex justify-content-between"
>
  <div>
    <p>
      2018
      :
    </p>
  </div>
  <div>
    <p>
      6
    </p>
  </div>
</div>
`;
